src/Groups/directproducts.jl

```diff
@@ -0,0 +1,266 @@
+################################################################################
+#
+#  Managing direct products
+#
+################################################################################
+
+export
+    directproduct,
+    embedding,
+    factorofdirectproduct,
+    homomorphism_of_semidirect_product,
+    isfull_direct_product,
+    projection,
+    semidirectproduct,
+    sub,
+    write_as_full
+
+
+"""
+    directproduct(G::S, H::T)
+Return the direct product of `G` and `H`, of type ``DirectProductOfGroups{S,T}``.
+"""
+function directproduct(G::S, H::T) where S<:GAPGroup where T<:GAPGroup
+   return DirectProductOfGroups(GAP.Globals.DirectProduct(G.X,H.X), G,H,true)
+end
+
+function factorofdirectproduct(G::DirectProductOfGroups, n::Base.Integer)
```

Comment:
Please add documentation comments for all new functions.

Also, please read up on `jldoctest `, as used e.g. in https://github.com/oscar-system/GAP.jl/pull/472 -- you can [read a description here](https://juliadocs.github.io/Documenter.jl/v0.7/man/doctests.html). In a nutshell, these are examples inside docstrings which *also* can be executed as tests. This has multiple advantages
- you are writing usage examples and tests at the same time, which is a clear win
- the tests are right next to the functions, making it much easier to verify each function has at least one test

So, please read up on these, then start using them. Make sure to *also* figure out how to execute these tests -- I think you might have to adjust `docs/make.jl` for that (change a `false` to a `true` somewhere, or so)

Reply:
@GDeFranceschi Concerning the execution of doctests, Documenter.jl supports two features:

- If the `doctest` argument of `makedocs` (in `docs/make.jl`) has the value `true` then the tests are executed whenever the documentation is built.
- One can call the function `Documenter.doctest` in a testfile, then the tests are executed whenever the package tests are run.

(Currently GAP.jl does both.)